src/getFlagsProxy.ts

```diff
@@ -60,15 +60,18 @@ function toFlagsProxy(ldClient: LDClient, flags: LDFlagSet, flagKeyMap: LDFlagKe
       const currentValue = Reflect.get(target, prop, receiver);
 
       // only process flag keys and ignore symbols and native Object functions
-      if (typeof prop === 'symbol' || !hasFlag(flagKeyMap, prop)) {
+      if (typeof prop === 'symbol' || (!hasFlag(flags, prop) && !hasFlag(flagKeyMap, prop))) {
         return currentValue;
       }
 
+      // check camelized map first before just using passed-in prop
+      const flagName = hasFlag(flagKeyMap, prop) ? flagKeyMap[prop] : prop;
+
       if (currentValue === undefined) {
         return;
       }
 
-      return ldClient.variation(flagKeyMap[prop], currentValue);
+      return ldClient.variation(flagName, currentValue);
```

Comment:
`flagName` should be renamed `originalFlagKey` to be accurate and only needs to be calculated after the `undefined` currentValue check:

```ts
      if (currentValue === undefined) {
        return;
      }

      const originalFlagKey = useCamelCaseFlagKeys ? flagKeyMap[prop] : prop;
      return ldClient.variation(originalFlagKey, currentValue);
```